ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/OpenFlowUtils.java

```diff
@@ -195,34 +196,35 @@ private static Flow createDefaultFlow() {
                                 .build();
     }
 
-    private static List<Flow> createVlanPassingFlows(String outputPort, int internalVlanId, String serviceName, List<Link> interswitchLinks) {
+    private static List<Flow> createVlanPassingFlows(String outputPort, int internalVlanId, int externalVlanId, String serviceName, List<Link> interswitchLinks) {
         return interswitchLinks.stream()
-                               .map(link -> createVlanPassingFlow(outputPort, link.getLinkId().getValue(), internalVlanId, serviceName))
+                               .map(link -> createVlanPassingFlow(outputPort, link.getLinkId().getValue(), internalVlanId, externalVlanId,serviceName))
                                .collect(Collectors.toList());
     }
 
-    private static Flow createVlanPassingFlow(String outputPort, String inputPort, int internalVlanId, String serviceName) {
+    private static Flow createVlanPassingFlow(String outputPort, String inputPort, int internalVlanId, int externalVlanId, String serviceName) {
         // Create list of actions and VLAN match
         List<Action> actions = new ArrayList<>();
-        Match vlanMatch;
         int actionOrder = 0;
 
-        vlanMatch = MatchUtils.createVlanMatch(internalVlanId, inputPort);
-
         actions.add(ActionUtils.createPopVlanAction(actionOrder++));
+        if(externalVlanId>=0){
```

Comment:
why do we need to check that? (as well in other places)

Reply:
When externalVlanId is -1 it means that it is port-base service and then no vlan is poped on input port and added on outport